sys-process/bottom/bottom-0.9.7.ebuild

```diff
@@ -0,0 +1,243 @@
+	redox_syscall@0.2.16
+	redox_syscall@0.3.5
+	redox_users@0.4.3
+	regex@1.9.4
+	regex-automata@0.3.7
+	regex-syntax@0.7.5
+	roff@0.2.1
+	rustc-demangle@0.1.23
+	rustix@0.37.23
+	rustix@0.38.9
+	ryu@1.0.15
+	same-file@1.0.6
+	scopeguard@1.2.0
+	serde@1.0.188
+	serde_derive@1.0.188
+	serde_json@1.0.105
+	serde_spanned@0.6.3
+	signal-hook@0.3.17
+	signal-hook-mio@0.2.3
+	signal-hook-registry@1.4.1
+	smallvec@1.11.0
+	starship-battery@0.8.2
+	static_assertions@1.1.0
+	strsim@0.9.3
+	strsim@0.10.0
+	syn@1.0.109
+	syn@2.0.32
+	sysctl@0.5.4
+	sysinfo@0.29.8
+	terminal_size@0.2.6
+	termtree@0.4.1
+	thiserror@1.0.47
+	thiserror-impl@1.0.47
+	time@0.3.36
+	time-core@0.1.2
+	time-macros@0.2.18
+	toml_datetime@0.6.3
+	toml_edit@0.19.14
+	typenum@1.16.0
+	unicode-ident@1.0.11
+	unicode-segmentation@1.10.1
+	unicode-width@0.1.10
+	uom@0.35.0
+	utf8parse@0.2.1
+	version_check@0.9.4
+	wait-timeout@0.2.0
+	walkdir@2.3.3
+	wasi@0.11.0+wasi-snapshot-preview1
+	winapi@0.3.9
+	winapi-i686-pc-windows-gnu@0.4.0
+	winapi-util@0.1.5
+	winapi-x86_64-pc-windows-gnu@0.4.0
+	windows@0.51.1
+	windows-core@0.51.1
+	windows-sys@0.48.0
+	windows-targets@0.48.5
+	windows_aarch64_gnullvm@0.48.5
+	windows_aarch64_msvc@0.48.5
+	windows_i686_gnu@0.48.5
+	windows_i686_msvc@0.48.5
+	windows_x86_64_gnu@0.48.5
+	windows_x86_64_gnullvm@0.48.5
+	windows_x86_64_msvc@0.48.5
+	winnow@0.5.14
+	wrapcenum-derive@0.4.0
+	zerocopy@0.7.35
+	zerocopy-derive@0.7.35
+"
+
+inherit bash-completion-r1 cargo
+
+DESCRIPTION="A graphical process/system monitor with a customizable interface"
+HOMEPAGE="https://github.com/ClementTsang/bottom"
+SRC_URI="${CARGO_CRATE_URIS}"
+
+LICENSE="0BSD Apache-2.0 Apache-2.0-with-LLVM-exceptions BSD-2 Boost-1.0 ISC MIT MPL-2.0 Unicode-DFS-2016 Unlicense ZLIB"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~ppc64"
+IUSE="+battery +gpu +zfs"
+
+# Rust packages ignore CFLAGS and LDFLAGS so let's silence the QA warnings
+QA_FLAGS_IGNORED="usr/bin/btm"
+
+PATCHES=(
+	"${FILESDIR}"/0.9.7-Bump-ahash-from-0.8.3-to-0.8.11.patch
+	"${FILESDIR}"/0.9.7-RUSTSEC-2024-0019-Update-mio-from-0.8.8-to-0.8.11.patch
+)
+
+src_prepare() {
+	# Stripping symbols should be the choice of the user.
+	sed -i '/strip = "symbols"/d' Cargo.toml || die "Unable to patch out symbol stripping"
+
+	eapply_user
+
+	default
+}
+
+src_configure() {
+	myfeatures=(
+		$(usev battery)
+		$(usev gpu)
+		$(usev zfs)
+	)
+
+	# This will turn on generation of shell completion scripts
+	export BTM_GENERATE=true
+
+	# https://github.com/ClementTsang/bottom/blob/bacaca5548c2b23d261ef961ee6584b609529567/Cargo.toml#L63
+	# fern and log features are for debugging only, so disable default features
+	cargo_src_configure $(usev !debug --no-default-features)
+}
+
+src_install() {
+	cargo_src_install
+
+	# Find generated shell completion files. btm.bash can be present in multiple dirs if we build
+	# additional features, so grab the first match only.
+	local BUILD_DIR="$(dirname $(find target -name btm.bash -print -quit || die) || die)"
```

Comment:
This doesn't sound like the global `BUILD_DIR` var, so make it lowercase to avoid accidental collision with something.

Reply:
Fixed in https://github.com/gentoo/gentoo/compare/1e795f19173d79b8ce9623162e752e194205150e..4926fde527c8a2fc13c8d9983918691107cbcf1c